lib/encrypted_sidekiq_redis.rb

```diff
@@ -0,0 +1,97 @@
+require 'redis'
+require 'gibberish'
+
+class EncryptedSidekiqRedis
+  attr_accessor :redis, :cipher
+
+  def initialize(opts)
+    self.redis = Redis.new(opts)
+    self.cipher = Gibberish::AES.new(Figaro.env.session_encryption_key)
+  end
+
+  def lpush(key, value)
+    super(key, encrypt_job(value))
+  end
+
+  def lpushx(key, value)
+    super(key, encrypt_job(value))
+  end
+
+  def rpush(key, value)
+    super(key, encrypt_job(value))
+  end
+
+  def lpop(key)
+    decrypt_job(super(key))
+  end
+
+  def rpop(key)
+    decrypt_job(super(key))
+  end
+
+  def blpop(*args)
+    queue, job = super(args)
+    [queue, decrypt_job(job)]
+  end
+
+  def brpop(*args)
+    queue, job = super(args)
+    [queue, decrypt_job(job)]
+  end
+
+  def zadd(key, *args)
+    ts, job = args
+    super(key, [ts, encrypt_job(job)])
+  end
+
+  def zrem(key, member)
+    # member must be removed from redis as-is (encrypted)
+    # but it is used elsewhere as if it was decrypted, so alter it in place.
+    ret = super(key, member)
+    if ret
+      decrypted_job = decrypt_job(member)
+      member.clear
+      member << decrypted_job
+    end
+    ret
+  end
+
+  def method_missing(meth, *args, &block)
+    #puts "redis #{meth} args:#{args.inspect}"
```

Comment:
Out of curiosity, what other redis methods does sidekiq end up using? `llen` maybe?